src/execution_plan/execution_plan.c

```diff
@@ -321,135 +336,150 @@ static AR_ExpNode** _OrderClause_GetExpressions(const AST *ast) {
-                                                         exps[i]->edgeLength->minHops,
-                                                         exps[i]->edgeLength->maxHops,
-                                                         g,
-                                                         ast);
-                        }
-                        else {
-                            op = NewCondTraverseOp(exps[i], ast);
-                        }
-                        Vector_Push(traversals, op);
-                    }
+                AlgebraicExpression **exps = AlgebraicExpression_From_QueryGraph(cc, ast, &expCount);
+
+                // Reorder exps, to the most performent arrangement of evaluation.
+                orderExpressions(exps, expCount, filter_tree);
+
+                AlgebraicExpression *exp = exps[0];
+                selectEntryPoint(exp, filter_tree);
+
+                // Create SCAN operation.
+                if(exp->src_node->label) {
+                    /* There's no longer need for the last matrix operand
+                     * as it's been replaced by label scan. */
+                    AlgebraicExpression_RemoveTerm(exp, exp->operand_count-1, NULL);
+                    op = NewNodeByLabelScanOp(exp->src_node, ast);                    
                 } else {
-                    AlgebraicExpression *exp = exps[expCount-1];
-                    selectEntryPoint(exp, filter_tree);
-                    // Create SCAN operation.
-                    if(exp->dest_node->label) {
-                        /* There's no longer need for the last matrix operand
-                         * as it's been replaced by label scan. */
-                        AlgebraicExpression_RemoveTerm(exp, exp->operand_count-1, NULL);
-                        op = NewNodeByLabelScanOp(exp->dest_node, ast);
-                        Vector_Push(traversals, op);
-                    } else {
-                        op = NewAllNodeScanOp(g, exp->dest_node, ast);
-                        Vector_Push(traversals, op);
+                    op = NewAllNodeScanOp(g, exp->src_node, ast);
+                }
+
+                Vector_Push(traversals, op);
+                _UpdateResolvedVariables(resolved, op);
+
+                for(int i = 0; i < expCount; i++) {
+                    exp = exps[i];
+                    if(exp->operand_count == 0) continue;
+                    
+                    /* Make sure expression source is already resolved.
+                     * TODO: it might be better to ensure beforehand 
+                     * that exp's source is already resolved, see 
+                     * traverse_order.c */
+                    if(raxFind(resolved,
+                        (unsigned char*)exp->src_node->alias,
+                        strlen(exp->src_node->alias)) == raxNotFound) {
+
+                        AlgebraicExpression_Transpose(exp);
+
+                        assert(raxFind(resolved,
```

Comment:
This assertion fails on the query:
`"MATCH (p:person)-[*]->(q:person) WHERE q.name = 'Roi Lipman' RETURN p, q"`

Reply:
Good catch!